js/search_bar.ts

```diff
@@ -0,0 +1,400 @@
+import { loadFonts } from "./utils";
+
+import { TabMode } from "./tab_panel";
+import { unsafeHTML } from "lit/directives/unsafe-html.js";
+
+export interface SearchTab {
+    search: string,
+    results: string[],
+    selected: string,
+    additional_html: string,
+}
+
+
+export interface SearchBarModel {
+    expanded: boolean;
+    tab_index: number;
+    name_address_model: string,
+    lat_lon_model: string,
+    dataset_model: string,
+}
+
+const DEBOUNCE_TIMEOUT = 500;
+
+const debounce = (callback: Function) => {
+    let timeoutId: number | undefined = undefined;
+    return (...args: any[]) => {
+        window.clearTimeout(timeoutId);
+        timeoutId = window.setTimeout(() => {
+            callback(...args);
+        }, DEBOUNCE_TIMEOUT);
+    };
+}
+
+export class SearchBar extends LitWidget<
+    SearchBarModel,
+    SearchBar
+> {
+    static get componentName() {
+        return `search-bar`;
+    }
+
+    static styles = [
+        legacyStyles,
+        materialStyles,
+        css`
+            .row {
+                display: flex;
+            }
+
+            .expand-button {
+                border-radius: 5px;
+                font-size: 16px;
+                height: 28px;
+                margin: 2px;
+                user-select: none;
+                width: 28px;
+            }
+
+            .hide {
+                display: none;
+            }
+
+            .expanded {
+                display: block; !important
+            }
+
+            input.search {
+                margin: 2px;
+                width: calc(100% - 4px);
+            }
+
+            ul.results {
+                list-style-type: none;
+                margin: 0;
+                margin-bottom: 4px;
+                max-width: 340px;
+                padding: 0;
+            }
+
+            label.result {
+                align-items: center;
+                display: flex;
+            }
+
+            .import-button, .reset-button {
+                margin: 0 2px 2px 2px;
+                padding: 0 8px;
+            }
+
+            .dataset-select {
+                margin-bottom: 2px;
+                margin-right: 2px;
+                max-width: 285px;
+            }
+
+            .additional-html-container {
+                max-height: 300px;
+                max-width: 340px;
+                overflow: auto;
+            }
+
+            .additional-html-container pre {
+                white-space: break-spaces;
+            }
+        `,
+    ];
+
+    modelNameToViewName(): Map<keyof SearchBarModel, keyof SearchBar> {
+        return new Map([
+            ["expanded", "expanded"],
+            ["tab_index", "tab_index"],
+            ["name_address_model", "name_address_model"],
+            ["lat_lon_model", "lat_lon_model"],
+            ["dataset_model", "dataset_model"],
+        ]);
+    }
+
+    @property()
+    expanded: boolean = false;
+
+    @property()
+    tab_index: number = 0;
```

Comment:
Consider using lowerCamelCase for the properties here. You'll have to rename the property names in `modelNameToViewName`.

Reply:
Good call! That cleans things up nicely. Done.